### PR TITLE
+poppler.freedesktop.org

### PR DESCRIPTION
**Pull request by jonchang**
also add the poppler-data package separately. (Namespacing is great!)

### CODE REVIEW

**Comment by jonchang** [comment]
~~Looks like we will need to package libopenjpeg and libtiff~~ done

**Comment by mxcl** [comment]
Added:
* #433
* #434

**Comment by jonchang** [comment]
Odd failure on darwin/x86-64. 

```
g-ir-scanner: link: gcc -o /opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib/tmp-introspectimjfx6ue/Poppler-0.18 /opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib/tmp-introspectimjfx6ue/Poppler-0.18.o -L. -Wl,-rpath,. -L/opt/poppler.freedesktop.org/src-v22.12.0/build_shared -Wl,-rpath,/opt/poppler.freedesktop.org/src-v22.12.0/build_shared -L/opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib -Wl,-rpath,/opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib -lpoppler-glib -lpoppler -L/opt/gnome.org/glib/v2.72.4/lib/pkgconfig/../../lib -L/opt/gnome.org/glib/v2.72.4/lib/pkgconfig/../../../../../gnu.org/gettext/v0/lib -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -lglib-2.0 -lintl -Wl,-rpath,/opt
dyld[49133]: Library not loaded: /usr/local/opt/freetype/lib/libfreetype.6.dylib
  Referenced from: <CF326A57-9BEF-336D-BC3B-029D12F45C90> /opt/cairographics.org/v1.16.0/lib/libcairo.2.dylib
  Reason: tried: '/usr/local/opt/freetype/lib/libfreetype.6.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/freetype/lib/libfreetype.6.dylib' (no such file), '/usr/local/opt/freetype/lib/libfreetype.6.dylib' (no such file), '/opt/poppler.freedesktop.org/src-v22.12.0/build_shared/libfreetype.6.dylib' (no such file), '/opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib/libfreetype.6.dylib' (no such file)
Command '['/opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib/tmp-introspectimjfx6ue/Poppler-0.18', '--introspect-dump=/opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib/tmp-introspectimjfx6ue/functions.txt,/opt/poppler.freedesktop.org/src-v22.12.0/build_shared/glib/tmp-introspectimjfx6ue/dump.xml']' died with <Signals.SIGABRT: 6>.
```

On darwin/aarch64:

```console
% otool -L libcairo.2.dylib
libcairo.2.dylib:
	@rpath/cairographics.org/v1.16.0/lib/libcairo.2.dylib (compatibility version 11603.0.0, current version 11603.0.0)
	@rpath/pixman.org/v0/lib/libpixman-1.0.40.0.dylib (compatibility version 41.0.0, current version 41.0.0)
	@rpath/freedesktop.org/fontconfig/v2/lib/libfontconfig.1.dylib (compatibility version 14.0.0, current version 14.0.0)
	@rpath/freetype.org/v2/lib/libfreetype.6.18.3.dylib (compatibility version 6.0.0, current version 6.18.3)
	@rpath/libpng.org/v1/lib/libpng16.16.dylib (compatibility version 56.0.0, current version 56.0.0)
	@rpath/zlib.net/v1/lib/libz.1.2.13.dylib (compatibility version 1.0.0, current version 1.2.13)
	/System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices (compatibility version 1.0.0, current version 61.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1953.255.0)
	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1690.3.3)
```

But on darwin/x86-64:

```console
% otool -L libcairo.dylib
libcairo.dylib:
	@rpath/cairographics.org/v1.16.0/lib/libcairo.2.dylib (compatibility version 11603.0.0, current version 11603.0.0)
	@rpath/pixman.org/v0/lib/libpixman-1.0.40.0.dylib (compatibility version 41.0.0, current version 41.0.0)
	@rpath/libpng.org/v1/lib/libpng16.16.dylib (compatibility version 56.0.0, current version 56.0.0)
	/usr/local/opt/freetype/lib/libfreetype.6.dylib (compatibility version 25.0.0, current version 25.3.0)
	@rpath/zlib.net/v1/lib/libz.1.2.13.dylib (compatibility version 1.0.0, current version 1.2.13)
	/usr/local/opt/libxrender/lib/libXrender.1.dylib (compatibility version 5.0.0, current version 5.0.0)
	/usr/local/opt/libx11/lib/libX11.6.dylib (compatibility version 11.0.0, current version 11.0.0)
	/usr/local/opt/libxext/lib/libXext.6.dylib (compatibility version 11.0.0, current version 11.0.0)
	/System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices (compatibility version 1.0.0, current version 56.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1856.105.0)
	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1557.3.2)
```

The build for intel macs seems to reference libraries that shouldn't exist.

**Comment by jonchang** [comment]
@jhheider any chance you could kick off a rebuild for intel darwin for cairographics.org? It seems to be picking up Homebrew X11 libraries that it shouldn't be using.

**Comment by jhheider** [comment]
Yup.

**Comment by jhheider** [comment]
Can confirm those paths aren't present on our self-hosted builder, so that's a consequence of GHA `macos-11`.

**Comment by jhheider** [comment]
Done: https://github.com/teaxyz/pantry.core/actions/runs/4319977717

**Comment by jhheider** [comment]
```sh
$ otool -l cairographics.org/v1.16.0/lib/libcairo.2.dylib | grep dylib
cairographics.org/v1.16.0/lib/libcairo.2.dylib:
         name @rpath/cairographics.org/v1.16.0/lib/libcairo.2.dylib (offset 24)
         name @rpath/pixman.org/v0/lib/libpixman-1.0.40.0.dylib (offset 24)
         name @rpath/freedesktop.org/fontconfig/v2/lib/libfontconfig.1.dylib (offset 24)
         name @rpath/freetype.org/v2/lib/libfreetype.6.18.3.dylib (offset 24)
         name @rpath/libpng.org/v1/lib/libpng16.16.dylib (offset 24)
         name @rpath/zlib.net/v1/lib/libz.1.2.13.dylib (offset 24)
         name /usr/lib/libSystem.B.dylib (offset 24)
```

**Comment by jonchang** [comment]
Tested the linux/x86-64 artifacts locally, it looks like there's an issue with relocation. Going to see what I can find out about what patchelf is doing (or not doing)

**Comment by jhheider** [comment]
We've been developing more issues recently. For many things, using PIE will fix it, but not for all things. Your expertise in the depths of binary formats is sorely appreciated.

**Comment by jonchang** [comment]
I note that we're using patchelf 1.17.0 still, but .2 released a little while ago and has a few fixes for out of bounds segment references.

**Comment by jhheider** [comment]
Triggering the build.

**Comment by jhheider** [comment]
Yeah, confirmed that it's definitely PIE.

File: projects/poppler.freedesktop.org/package.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  - bin/pdfunite
+
+# TODO: Add in libopenjpeg, libtiff
+dependencies:
+  boost.org: '>=1.58.0'
+  cairographics.org: '>=1.16.0'
+  curl.se: '*'
+  freedesktop.org/fontconfig: '>=2.13'
+  freetype.org: '>=2.10'
+  gnome.org/glib: '>=2.64'
```
Comment (mxcl) [inline projects/poppler.freedesktop.org/package.yml:30]: I think `^2.64` is more likely. Using `>=` includes a potential future v3. Same for the rest.

File: projects/poppler.freedesktop.org/package.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  libjpeg-turbo.org: '*'
+  libpng.org: '*'
+  openjpeg.org: '*'
+  poppler.freedesktop.org/poppler-data: '*'
```
Comment (mxcl) [inline projects/poppler.freedesktop.org/package.yml:34]: I would have preferred `poppler.freedesktop.org/data` but the beauty of namespacing is this is contained so not important.

**Review by mxcl — APPROVED** [review]
Biggest dep tree so far! Great work.

**Pull request merged**